### PR TITLE
feat: Add workflow dispatch to prisma/prisma-fmt-wasm

### PR DESCRIPTION
**Pull request by janpio**
Includes both version string of just released packages but also the original engine commit hash as input.

### CODE REVIEW

**Review by tomhoule — APPROVED** [review]
:+1:

**Comment by janpio** [comment]
Waiting with merge until the equivalent workflow is implemented on the prisma-fmt-wasm side.

**Comment by Jolg42** [comment]
The repo is https://github.com/prisma/prisma-fmt-wasm

**Comment by janpio** [comment]
As we are publishing to different dist tags here, we also need to do the same over in the wasm repo: https://www.npmjs.com/package/@prisma/engines
image
I'll add that here as an additional input.

File: .github/workflows/publish-engines.yml [hunk 1]
```diff
@@ -38,14 +38,23 @@ jobs:
           GITHUB_EVENT_CLIENT_PAYLOAD: ${{ toJson(github.event.client_payload) }}
 
       - name: Workflow dispatch to prisma/prisma for version update
-        if: ${{ steps.publish_script.outputs.new_prisma_version }}
+        # Only automatically trigger update for `latest`!
+        if: steps.publish_script.outputs.npm_dist_tag == 'latest'
```
Comment (janpio) [inline .github/workflows/publish-engines.yml:42]: `new_prisma_version` was only set in publish.ts if `npmDistTag == latest`.
This is now changed to _always_ be set, but the Npm dist tag itself is also made available as a variable.
Use that new variable instead to achieve the same result.

File: .github/workflows/publish-engines.yml [hunk 2]
```diff
@@ -38,14 +38,23 @@ jobs:
...
           token: ${{ secrets.BOT_TOKEN }}
           inputs: '{ "version": "${{ steps.publish_script.outputs.new_prisma_version }}" }'
 
+      - name: Workflow dispatch to prisma/prisma-fmt-wasm for version update
+        uses: benc-uk/workflow-dispatch@v1
+        with:
+          workflow: new-engines-version
+          repo: prisma/prisma-fmt-wasm
+          token: ${{ secrets.BOT_TOKEN }}
+          inputs: '{ "engineHash": "${{ github.event.client_payload.commit }}", "enginesWrapperVersion": "${{ steps.publish_script.outputs.new_prisma_version }}", "npmDistTag": "${{ steps.publish_script.outputs.npm_dist_tag }}" }'
```
Comment (janpio) [inline .github/workflows/publish-engines.yml:56]: new dispatch to `prisma-fmt-wasm` repo on the `new-engines-version` workflow, with 3 input values: `engineHash`, `enginesWrapperVersion`, and `npmDistTag`.

File: scripts/publish.ts [hunk 3]
```diff
@@ -55,6 +55,12 @@ async function main(dryRun = false) {
   console.log(`${chalk.bold('New version')}   ${newVersion}`)
   console.log(`${chalk.bold('Npm Dist Tag')}  ${npmDistTag}\n`)
 
+  console.log(`Printing values for workflow dispatch: ${newVersion}, ${npmDistTag}`)
+  // This special log makes values avaliable in Github Actions
+  // Read: https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#using-workflow-commands-to-access-toolkit-functions
+  console.log(`::set-output name=new_prisma_version::${newVersion}`)
+  console.log(`::set-output name=npm_dist_tag::${npmDistTag}`)
```
Comment (janpio) [inline scripts/publish.ts:62]: always set variables instead of only on `npmDistTag == latest` so they are always available to use and make decisions.

**Comment by janpio** [comment]
This is now updated and ready for review again.

Goal of the changes: Only add an additional workflow dispatch in the other repo with 3 values as input. Existing behavior should stay unchanged.

File: scripts/publish.ts [hunk 4]
```diff
@@ -55,6 +55,12 @@ async function main(dryRun = false) {
   console.log(`${chalk.bold('New version')}   ${newVersion}`)
   console.log(`${chalk.bold('Npm Dist Tag')}  ${npmDistTag}\n`)
 
+  console.log(`Printing values for workflow dispatch: ${newVersion}, ${npmDistTag}`)
```
Comment (Jolg42) [inline scripts/publish.ts:58]:
```suggestion
```

It's a duplicate of the above

**Review by Jolg42 — APPROVED** [review]
Nice! Looks good!

**Comment by janpio** [comment]
Thanks for the review. We will wait until after the release tomorrow to merge, to not potentially create additional work to fix the pipeline here.

**Review by millsp — APPROVED** [review]

**Pull request merged**